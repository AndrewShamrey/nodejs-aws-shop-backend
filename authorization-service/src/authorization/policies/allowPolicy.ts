import { APIGatewayAuthorizerResult } from 'aws-lambda';

export const allowPolicy = (
  principalId = 'apigateway.amazonaws.com',
  context?: Record<string, string>,
): APIGatewayAuthorizerResult => ({
  principalId,
  context,
  policyDocument: {
    Version: '2012-10-17',
    Statement: [
      {
        Action: 'execute-api:Invoke',
        Effect: 'Allow',
        Resource: '*',
      },
    ],
  },
});
