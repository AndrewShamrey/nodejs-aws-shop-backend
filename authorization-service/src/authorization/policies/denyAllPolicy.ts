import { APIGatewayAuthorizerResult } from 'aws-lambda';

export const denyAllPolicy = (): APIGatewayAuthorizerResult => ({
  principalId: '*',
  policyDocument: {
    Version: '2012-10-17',
    Statement: [
      {
        Action: '*',
        Effect: 'Deny',
        Resource: '*',
      },
    ],
  },
});
