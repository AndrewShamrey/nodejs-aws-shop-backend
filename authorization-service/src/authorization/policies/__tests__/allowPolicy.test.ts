import { allowPolicy } from 'authorization/policies/allowPolicy';

describe('allowPolicy', () => {
  it('should return valid allow policy object', () => {
    expect(allowPolicy()).toMatchSnapshot();
    expect(allowPolicy('test-id')).toMatchSnapshot();
  });
});
