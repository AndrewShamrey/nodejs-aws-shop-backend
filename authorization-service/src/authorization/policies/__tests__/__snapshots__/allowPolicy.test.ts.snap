// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allowPolicy should return valid allow policy object 1`] = `
Object {
  "context": undefined,
  "policyDocument": Object {
    "Statement": Array [
      Object {
        "Action": "execute-api:Invoke",
        "Effect": "Allow",
        "Resource": "*",
      },
    ],
    "Version": "2012-10-17",
  },
  "principalId": "apigateway.amazonaws.com",
}
`;

exports[`allowPolicy should return valid allow policy object 2`] = `
Object {
  "context": undefined,
  "policyDocument": Object {
    "Statement": Array [
      Object {
        "Action": "execute-api:Invoke",
        "Effect": "Allow",
        "Resource": "*",
      },
    ],
    "Version": "2012-10-17",
  },
  "principalId": "test-id",
}
`;
